tests/serverpod_test_server/test_integration/database_operations/transactions/transaction_test.dart

```diff
@@ -131,16 +133,30 @@ void main() async {
           'ROLLBACK TO SAVEPOINT savepoint1;',
           transaction: transaction,
         );
-
+        await session.db.transaction((transaction2) async {
+          await session.db.unsafeExecute(
+            'SAVEPOINT savepoint2;',
+            transaction: transaction2,
+          );
+          await UniqueData.db
+              .insertRow(session, data4, transaction: transaction2);
+          await session.db.unsafeExecute(
+            'ROLLBACK TO SAVEPOINT savepoint2;',
+            transaction: transaction2,
+          );
+          await UniqueData.db
+              .insertRow(session, data5, transaction: transaction);
+        });
         await UniqueData.db.insertRow(session, data3, transaction: transaction);
       },
     );
 
     var fetchedData = await UniqueData.db.find(session);
 
     expect(fetchedData, isNotEmpty);
-    expect(fetchedData, hasLength(2));
+    expect(fetchedData, hasLength(3));
```

Comment:
@SandPod I thought I'd extend this test to include the nested case, since it does prove an important aspect. It's not directly related to the fix in this PR, but maybe we can sneak it in here? Or would you like me to open a separate PR for it?

Reply:
Makes total sense to add a new test for this, but how about adding it as a separate scenario? Maybe we could have a discussion around how the nested transaction test could be structured to make it as simple as possible but still capture the essence of it.